db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSTABLESRowFactory.java

```diff
@@ -52,661 +52,664 @@
-	 */
-	public SystemColumn[]	buildColumnList()
-        throws StandardException
-	{
-        return new SystemColumn[] {
-                SystemColumnImpl.getUUIDColumn("TABLEID", false),
-                SystemColumnImpl.getIdentifierColumn("TABLENAME", false),
-                SystemColumnImpl.getIndicatorColumn("TABLETYPE"),
-                SystemColumnImpl.getUUIDColumn("SCHEMAID", false),
-                SystemColumnImpl.getIndicatorColumn("LOCKGRANULARITY"),
-                SystemColumnImpl.getIdentifierColumn("VERSION",true),
-                SystemColumnImpl.getColumn("COLSEQUENCE", Types.INTEGER, false),
-                SystemColumnImpl.getColumn("DELIMITED", Types.VARCHAR,  true),
-                SystemColumnImpl.getColumn("ESCAPED", Types.VARCHAR, true),
-                SystemColumnImpl.getColumn("LINES", Types.VARCHAR, true),
-                SystemColumnImpl.getColumn("STORED", Types.VARCHAR, true),
-                SystemColumnImpl.getColumn("LOCATION", Types.VARCHAR, true),
-                SystemColumnImpl.getColumn("COMPRESSION", Types.VARCHAR, true),
-                SystemColumnImpl.getColumn("IS_PINNED", Types.BOOLEAN, false),
+        //TODO -sf- place version into tuple descriptor
+        DataValueDescriptor versionDescriptor = row.getColumn(SYSTABLES_VERSION);
+
+        DataValueDescriptor delimitedDVD = row.getColumn(SYSTABLES_DELIMITED_BY);
+        DataValueDescriptor escapedDVD = row.getColumn(SYSTABLES_ESCAPED_BY);
+        DataValueDescriptor linesDVD = row.getColumn(SYSTABLES_LINES_BY);
+        DataValueDescriptor storedDVD = row.getColumn(SYSTABLES_STORED_AS);
+        DataValueDescriptor locationDVD = row.getColumn(SYSTABLES_LOCATION);
+        DataValueDescriptor compressionDVD = row.getColumn(SYSTABLES_COMPRESSION);
+        // NOT USED ANYMORE, for backward compatibility only
+        @Deprecated
+        DataValueDescriptor isPinnedDVD = row.getColumn(SYSTABLES_IS_PINNED);
+        DataValueDescriptor purgeDeletedRowsDVD = row.getColumn(SYSTABLES_PURGE_DELETED_ROWS);
+        DataValueDescriptor minRetentionPeriodDVD = row.getColumn(SYSTABLES_MIN_RETENTION_PERIOD);
+
+        // RESOLVE - Deal with lock granularity
+        tabDesc = ddg.newTableDescriptor(tableName, schema, tableTypeEnum, lockGranularity.charAt(0),
+                row.getColumn(SYSTABLES_COLUMN_SEQUENCE).getInt(),
+                delimitedDVD != null ? delimitedDVD.getString() : null,
+                escapedDVD != null ? escapedDVD.getString() : null,
+                linesDVD != null ? linesDVD.getString() : null,
+                storedDVD != null ? storedDVD.getString() : null,
+                locationDVD != null ? locationDVD.getString() : null,
+                compressionDVD != null ? compressionDVD.getString() : null,
+                isPinnedDVD.getBoolean(),
+                purgeDeletedRowsDVD.getBoolean(),
+                minRetentionPeriodDVD.getLong()
```

Comment:
We may consider minRetentionPeriodDVD as NULL here for rows upgrade from lower versions.

Reply:
done.